sequencer/src/l1_client.rs

```diff
@@ -88,167 +85,57 @@ impl Committable for L1BlockInfo {
-                // maintaining our invariants (increasing L1 block numbers) over UX.
-                tracing::warn!(
-                    "got L1 head {head} which is not newer than previous head {}",
-                    snapshot.head
-                );
-                continue;
-            }
-
-            // A new block has been produced. This happens fairly rarely, so it is now ok to poll to
-            // see if a new block has been finalized.
-            let finalized = match get_finalized_block(&rpc, finalized_tag).await {
-                Ok(finalized) => Some(finalized),
-                Err(err) => {
-                    tracing::error!("failed to get finalized block: {err}");
-
-                    // If we cannot get the finalized block for any reason, don't let this stop us
-                    // from updating the L1 head. By returning `None` here, we will proceed to
-                    // update the head but not the finalized block, and then we will reset the
-                    // connection, in the hopes that this fixes whatever went wrong with
-                    // `get_finalized_block`.
-                    None
+    /// Proxy to `Provider.get_block_number`.
+    async fn get_block_number(&self) -> u64 {
+        loop {
+            match self.provider.get_block_number().await {
+                Ok(n) => return n.as_u64(),
+                Err(e) => {
+                    tracing::warn!("Blocknumber error: {}", e);
+                    sleep(self.retry_delay).await;
                 }
-            };
-
-            // Update the snapshot.
-            let mut snapshot = RwLockUpgradableReadGuard::upgrade(snapshot).await;
-            snapshot.head = head;
-            if let Some(finalized) = finalized {
-                snapshot.finalized = finalized;
             }
-
-            // Drop our exclusive lock before logging anything.
-            let snapshot = RwLockWriteGuard::downgrade(snapshot);
-            tracing::info!("updated L1 snapshot to {:?}", *snapshot);
-
-            // If we encountered an error in `get_finalized_block`, reset the connection now that we
-            // have updated what we can.
-            if finalized.is_none() {
-                tracing::warn!("resetting connection due to error in get_finalized_block");
-                break;
+        }
+    }
+    /// Proxy to `get_finalized_block`.
+    async fn get_finalized_block(&self) -> Option<L1BlockInfo> {
+        loop {
+            match get_finalized_block(&self.provider).await {
+                Ok(block) => return block,
+                Err(e) => {
+                    tracing::warn!("Finalized block error: {}", e);
+                    sleep(self.retry_delay).await;
+                }
             }
         }
-
-        tracing::error!("L1 block stream ended unexpectedly, trying to re-establish");
     }
 }
 
-async fn get_finalized_block<P: JsonRpcClient>(
+pub async fn get_finalized_block<P: JsonRpcClient>(
```

Comment:
Why does this need to be `pub`? Seems like anyone outside this module should just be calling `get_snapshot` on the client

Reply:
`pub` removed in ef28a828d766970c14c8d2b42b1b2afbadd4347e